hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/auth/Credentials.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.auth;
+
+import java.io.Serializable;
+
+import org.hawkular.cmdgw.api.Authentication;
+
+/**
+ * A serializable and immutable variant of {@link Authentication}.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class Credentials implements Serializable {
+    /**  */
+    private static final long serialVersionUID = 5468353L;
+    private final String password;
+    private final String username;
+
+    public Credentials(String username, String password) {
+        super();
+        this.username = username;
+        this.password = password;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Credentials other = (Credentials) obj;
+        if (password == null) {
+            if (other.password != null)
+                return false;
+        } else if (!password.equals(other.password))
+            return false;
+        if (username == null) {
+            if (other.username != null)
+                return false;
+        } else if (!username.equals(other.username))
+            return false;
+        return true;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((password == null) ? 0 : password.hashCode());
+        result = prime * result + ((username == null) ? 0 : username.hashCode());
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "Credentials [password=" + password + ", username=" + username + "]";
```

Comment:
Passwords should never be printed to logs or things like that. So, it's a good idea to remove the password from the `toString()` method.